### PR TITLE
Exempt Groovy 4 Gradleception from reporting cache misses

### PR DESCRIPTION
**Pull request by hegyibalint**
Should except the [Gradleception - Groovy 4.x Java8 Linux](https://builds.gradle.org/admin/editBuild.html?id=buildType:Gradle_Master_Check_GradleceptionWithGroovy4) build configuration from causing cache misses.

### CODE REVIEW

**Comment by hegyibalint** [comment]
@bot-gradle test RFM

File: build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.cache-miss-monitor.gradle.kts [hunk 1]
```diff
@@ -97,7 +97,8 @@ fun Project.isExpectedCompileCacheMiss() =
         "Check_CompileAllBuild",
         "Component_GradlePlugin_Performance_PerformanceLatestMaster",
         "Component_GradlePlugin_Performance_PerformanceLatestReleased",
-        "Check_Gradleception"
+        "Check_Gradleception",
+        "Gradle_Master_Check_GradleceptionWithGroovy4"
```
Comment (wolfs) [inline build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.cache-miss-monitor.gradle.kts:101]:
```suggestion
        "GradleceptionWithGroovy4"
```

To make this work with multiple branch projects in Teamcity we leave out the branch prefix here, and only check whether the given String is a suffix of the configuration name. See further down.

Comment (hegyibalint) [reply build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.cache-miss-monitor.gradle.kts:101]: Fair, the others are also without the project specifier.

**Review by wolfs — APPROVED** [review]
LGTM!

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Pull request merged**